### DISCUSSION TITLE
[Rdkafka] How to use the instrumentation with forks of the original gem that have a different gem name?

### DISCUSSION THREAD

**Discussion by piotaixr** [open-telemetry/opentelemetry-ruby-contrib#725] [Q&A] [answered]
Hi all,

The Karafka ecosystem uses the Rdkafka gem under the hood, but a fork of it with patches that are not yet upstreamed to the main rdkafka repo.
The rdkafka fork is available with the karafka-rdkafka gem, and the project do not include the original repo. When one tries to use the rdkafka instrumentation gem with it, it fails to load with the following error message:
`OpenTelemetry error: Instrumentation: OpenTelemetry::Instrumentation::Rdkafka unhandled exception during install: undefined method 'version' for nil:NilClass`

This happens because the way to check the rdkafka version available is done through the `Gem.loaded_specs` attribute.

My question is then: What would be the best way to use the instrumentation with this fork?
I see two possibilities:
- Changing the instrumentation code to be like the following so it do not depend on the gem but on the constant (like the `present` method does to see if the instrumented library is available in the environment)
```ruby
        compatible do
          Gem::Version.new(::Rdkafka::VERSION) >= MINIMUM_VERSION
        end
```
- Creating a whole new instrumentation for this specific fork.

Thanks for your answers!

**Comment by arielvalentin** [comment, marked as answer]
@piotaixr Please open a PR to use the `Rdkafka::VERSION` constant. It looks like we missed this when migrating away from using `Gem.loaded_specs`.

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/rdkafka/lib/opentelemetry/instrumentation/rdkafka/instrumentation.rb#L15
